src/System.Linq.Expressions/src/System/Runtime/CompilerServices/ReadOnlyCollectionBuilder.cs

```diff
@@ -23,7 +25,7 @@ public sealed class ReadOnlyCollectionBuilder<T> : IList<T>, System.Collections.
         private int _version;
 
         [NonSerialized]
-        private Object _syncRoot;
+        private object _syncRoot;
```

Comment:
> since this is not a wrapper on top of a publicly accessible collection.

This type itself is public, though. (I'm fine with the change you suggest, I'm just not understanding the argument made in support of it.)